### ISSUE TITLE
set-preference throws XML parsing error when setting high value for Text Facet's choice count limit

### ISSUE THREAD

**Issue by thadguidry** [OpenRefine/OpenRefine#1395]
Using Firefox latest

1. Added Text Facet on column with over 2 million rows.
2. Set choice count limit value to 1960000
3. Errors below

```
The page was reloaded, because the character encoding declaration of the HTML document was not found when prescanning the first 1024 bytes of the file. The encoding declaration needs to be moved to be within the first 1024 bytes of the file.
project:36
JQMIGRATE: Logging is active
project-bundle.js:10335:2

Synchronous XMLHttpRequest on the main thread is deprecated because of its detrimental effects to the end user’s experience. For more help http://xhr.spec.whatwg.org/
project-bundle.js:9594:5

Source map error: request failed with status 404
Resource URL: http://127.0.0.1:3333/project-bundle.js
Source Map URL: underscore-min.map
[Learn More]

XML Parsing Error: not well-formed
Location: http://127.0.0.1:3333/command/core/set-preference
Line Number 1, Column 1:
set-preference:1:1

Error: Script terminated by timeout at:
ListFacet.prototype._reSortChoices@http://127.0.0.1:3333/project-bundle.js:32528:3
ListFacet.prototype.updateState@http://127.0.0.1:3333/project-bundle.js:32518:5
BrowsingEngine.prototype.update/<@http://127.0.0.1:3333/project-bundle.js:31356:9
fire@http://127.0.0.1:3333/project-bundle.js:3121:10
fireWith@http://127.0.0.1:3333/project-bundle.js:3233:7
done@http://127.0.0.1:3333/project-bundle.js:9277:5
callback@http://127.0.0.1:3333/project-bundle.js:9687:8
```

**Comment by thadguidry** [comment]
Actually, upon waiting and waiting.. eventually I see another error...

**allocation size overflow**
project-bundle.js:32795:33
ListFacet.prototype._update
http://127.0.0.1:3333/project-bundle.js:32795:33
ListFacet.prototype.updateState
http://127.0.0.1:3333/project-bundle.js:32524:3
BrowsingEngine.prototype.update/<
http://127.0.0.1:3333/project-bundle.js:31356:9
fire
http://127.0.0.1:3333/project-bundle.js:3121:10
fireWith
http://127.0.0.1:3333/project-bundle.js:3233:7
done
http://127.0.0.1:3333/project-bundle.js:9277:5
callback
http://127.0.0.1:3333/project-bundle.js:9687:8

**Comment by thadguidry** [comment]
We know we don't deal with large facets very well already in #672

**Comment by tfmorris** [comment]
This doesn't seem high priority to me since it's an attempt to push beyond recommend boundaries. One easy way to fix the "bug" would be to enforce a reasonable upper limit on the text facet choice count.

**Comment by thadguidry** [comment]
Oh wow, this issue is still around?  Let's close it.  This was a limit test in actuality that I documented as an issue so that we had the capture of the actual error component.  This issue just needed a glance over by you, Jacky, or Antonin at the time and could have been closed as "Will not fix" long long ago.

So doing that now.  So to catch up from your time away, you only have about 400+ more issues to review Tom! ;-)  jk
